js/auth/AuthAdapterTB.js

```diff
@@ -27,6 +27,9 @@ AuthAdapter.prototype.load = function(cb) {
 			authToken = data.token;
 			decodedToken = lthis.decodeToken(data.token);
 			lthis.user.email = decodedToken.sub;
+			var now = new Date();
+			lthis.tokenExpirationTime = new Date(
+				now.getTime() + (data.duration - 10) * 1000);			
```

Comment:
Maybe `data.duration * 0.8` to avoid negative results ?

Reply:
Right! This is a cut and paste from the "cumulus-front" code so please consider also updating it.